Python/Product/PythonTools/PythonTools/Project/PythonTestPropertyPageView.xaml

```diff
@@ -0,0 +1,119 @@
+             FontSize="{DynamicResource VsFont.EnvironmentFontSize}"
+             d:DesignHeight="400" d:DesignWidth="400">
+    <UserControl.Resources>
+        <ResourceDictionary>
+            <Style TargetType="{x:Type Label}">
+                <Setter Property="Margin" Value="4" />
+            </Style>
+
+            <Style TargetType="{x:Type TextBox}">
+                <Setter Property="Margin" Value="4" />
+                <Setter Property="VerticalContentAlignment" Value="Center"/>
+            </Style>
+
+            <Style TargetType="{x:Type ComboBox}">
+                <Setter Property="Margin" Value="4" />
+                <Setter Property="VerticalContentAlignment" Value="Center"/>
+            </Style>
+
+            <DataTemplate x:Key="FrameworkNone">
+            </DataTemplate>
+            
+            <DataTemplate x:Key="FrameworkPytest">
+            </DataTemplate>
+            
+            <DataTemplate x:Key="FrameworkUnittest">
+                <Grid>
+                    <Grid.ColumnDefinitions>
+                        <ColumnDefinition Width="Auto" SharedSizeGroup="LabelSizeGroup"/>
+                        <ColumnDefinition Width="*"/>
+                    </Grid.ColumnDefinitions>
+                    <Grid.RowDefinitions>
+                        <RowDefinition/>
+                        <RowDefinition/>
+                    </Grid.RowDefinitions>
+                    <Label
+                        Name="RootDirectoryLabel"
+                        Grid.Row="0"
+                        Grid.Column="0"
+                        Content="{x:Static common:Strings.PythonTestPropertyPageUnittestRootDirectoryLabel}"/>
+                    <TextBox
+                        Grid.Row="0"
+                        Grid.Column="1"
+                        Text="{Binding Path=UnitTestRootDirectory, UpdateSourceTrigger=PropertyChanged}"
+                        AutomationProperties.LabeledBy="{Binding ElementName=RootDirectoryLabel}"
+                        AutomationProperties.HelpText="{x:Static common:Strings.PythonTestPropertyPageUnittestRootDirectoryHelpText}"/>
+                    <Label
+                        Name="PatternLabel"
+                        Grid.Row="1"
+                        Grid.Column="0"
+                        Content="{x:Static common:Strings.PythonTestPropertyPageUnittestPatternLabel}"/>
+                    <TextBox
+                        Grid.Row="1"
+                        Grid.Column="1"
+                        Text="{Binding Path=UnitTestPattern, UpdateSourceTrigger=PropertyChanged}"
+                        AutomationProperties.LabeledBy="{Binding ElementName=PatternLabel}"
+                        AutomationProperties.HelpText="{x:Static common:Strings.PythonTestPropertyPageUnittestPatternHelpText}"/>
+                </Grid>
+            </DataTemplate>
+        </ResourceDictionary>
+    </UserControl.Resources>
+
+    <StackPanel Grid.IsSharedSizeScope="True">
```

Comment:
do you really need this outter stackpanel?

Reply:
I need it because I want to have the framework specific content below the framework label/combo box, and you can only have one element at the root, so a container is needed.

There's always several ways to do anything in WPF ;)

I could have placed the content control as a 2-column spanning second row in the grid. But I feel it's simpler with the stack panel.